diff --git a/lib/src/components/Preview.tsx b/lib/src/components/Preview.tsx
index c6655d1e6bbecee4bfb73fc9cb531595a1a5cc1a..e5e01f9f3121fd9073e52e13ebbad6c5882ba65e 100644
--- a/lib/src/components/Preview.tsx
+++ b/lib/src/components/Preview.tsx
@@ -1,6 +1,6 @@
 import React, { useEffect, useState } from 'react';
 import { ImageBackground, View } from 'react-native';
-import Animated, { useAnimatedStyle, useDerivedValue, useSharedValue } from 'react-native-reanimated';
+import Animated, { ReduceMotion, useAnimatedStyle, useDerivedValue, useSharedValue, withTiming } from 'react-native-reanimated';
 
 import colorKit from '../colorKit/index';
 import usePickerContext from '../AppContext';
@@ -36,8 +36,9 @@ export function Preview({
     return contrast < 4.5 ? '#000000' : '#ffffff';
   }, [alphaValue, value]);
 
+  const displayedTextColor = useSharedValue<'#000000' | '#ffffff'>('#ffffff');
   const textColor = useSharedValue<'#000000' | '#ffffff'>('#ffffff');
-  const textColorStyle = useAnimatedStyle(() => ({ color: textColor.value }), [textColor]);
+  const textColorStyle = useAnimatedStyle(() => ({ color: displayedTextColor.value }), [displayedTextColor]);
 
   const previewColor = useSharedValue('#ffffff');
   const previewColorStyle = useAnimatedStyle(() => ({ backgroundColor: previewColor.value }), [previewColor]);
@@ -55,7 +56,12 @@ export function Preview({
     const compareColor2 = textColor.value === '#000000' ? { h: 0, s: 0, v: 0 } : { h: 0, s: 0, v: 100 };
     const contrast = colorKit.runOnUI().contrastRatio(compareColor1, compareColor2);
     const reversedColor = textColor.value === '#ffffff' ? '#000000' : '#ffffff';
-    textColor.value = contrast < 4.5 ? reversedColor : textColor.value;
+
+    if (contrast < 4.5) {
+      textColor.value = reversedColor
+      displayedTextColor.value = withTiming(reversedColor, { reduceMotion: ReduceMotion.Never })
+    }
+
   }, [hueValue, saturationValue, brightnessValue, alphaValue]);
 
   return (
